include_server/parse_command.py

```diff
@@ -119,6 +119,24 @@ def _RaiseNotImplemented(name, comment=''):
 for key in CPP_OPTIONS_MAYBE_TWO_WORDS.keys():
   assert key[1] in CPP_OPTIONS_MAYBE_TWO_WORDS_FIRST_LETTERS
 
+PATH_EXPR='[/a-zA-Z_0-9.]+' # regular expression for a partial file path
+
+# These are the cpp options that require regular expressions, m is Match.
+CPP_OPTIONS_REGULAR_EXPRESSIONS = {
+  '-Wa,(%s\.s)' % PATH_EXPR:     lambda ps, m: ps.include_files.append(m.group(1)),
+  '-Wa,\[(%s\.s)\]' % PATH_EXPR: lambda ps, m: ps.include_files.append(m.group(1)),
+  '-Wa,(%s\.S)' % PATH_EXPR:     lambda ps, m: ps.include_files.append(m.group(1)),
+  '-Wa,\[(%s\.S)\]' % PATH_EXPR: lambda ps, m: ps.include_files.append(m.group(1)),
+}
```

Comment:
It may not be necessary to support .S files as, by convention, they need to be run through the preprocessor and the -Wa argument will skip the pre-processor. 

In the kernel, macro.s is already generated from macro.S in a different step.

Reply:
Good to know, we do run cpp on the server (in pump mode) but if it doesn't handle it anyway...